### PR TITLE
Migrate sanity-plugin-table to Sanity V3

### PR DESCRIPTION
**Pull request by davydog187**
Follows https://beta.sanity.io/docs/platform/studio/v2-to-v3/plugins

## Changes
1. Removes the [`rowType` feature ](https://github.com/rdunk/sanity-plugin-table#configuration)that was only needed when the row type was renamed from version 1 -> 2
2. Removes all the custom css in favor of inline styles and using Sanity UI components. It is not a 1-1 change but generally looks good
3. Refactors some of the file locations. All the schemas are now in the `src/index.ts` inline with the `definePlugin` definition
4. Renders "Empty Table" when the table has not been created yet
5. Removes all references to [`React.FunctionComponent`](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/)
6. exports the plugin as `{ table }`

Screen Shot 2022-11-09 at 6 06 22 PM

## Outstanding issues
1. When the preview is focused, its not updating the styles of the Textbox and table preview with the focus background color. According to [the documentation](https://beta.sanity.io/docs/learn/custom-input-component/importing-an-input-element#adding-necessary-properties-and-making-sure-the-field-gets-focus), the `elementProps` need to be forwarded to the component, but they don't appear to be passed to the `preview` component. I asked a question in the [sanity input component discussion](https://github.com/sanity-io/sanity/discussions/3527#discussioncomment-4102150) on how to do this correctly, hopefully we get an answer
Screen Shot 2022-11-09 at 6 07 47 PM

### CODE REVIEW

File: package.config.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  dist: 'lib',
+  minify: true,
+  legacyExports: true,
+  // Remove this block to enable strict export validation
+  extract: {
+    rules: {
+      'ae-forgotten-export': 'off',
+      'ae-incompatible-release-tags': 'off',
+      'ae-internal-missing-underscore': 'off',
+      'ae-missing-release-tag': 'off',
```
Comment (davydog187) [inline package.config.ts:13]: I copied this from [sanity-plugin-media@v3](https://github.com/sanity-io/sanity-plugin-media/blob/v3/package.config.ts). Not sure if we want this rules enabled

**Comment by snorrees** [comment]
This is awesome stuff! Trying to determine what to do about ownership and naming, and will do a proper review shortly.

In the meantime: We have been doing some work lately on plugin-kit to ship an opinionated used-by-sanity semantic-release driven Github workflow as an opt-in feature. 

To test it (I would recommend this as part of a PR anyway ;) )
* update plugin-kit to ^2.1.13, 
* commit any changes (so you have no local diff)
* in your repo root run `npx @sanity/plugin-kit inject --preset-only --preset semver-workflow`
* follow the instructions in command log
* inspect the diff to see what changed:
   * husky setup to enforce conventional-commits
   * semantic-release with sanity-opinionated dependency
   * Full github workflow with click-to-release support via Github actions

Incidentally I will talk about this that in tomorrows [Sanity meetup](https://www.meetup.com/meetup-group-dvjyrjdv/events/289456759/) (shameless plug ;))

File: package.config.ts (hunk 1 above)
Comment (snorrees) [reply package.config.ts:13]: No rush to enable these atm, but they will come into play once we get around to shipping automated docs pages based on ts-docs in plugin-kit.

**Comment by snorrees** [comment]
Quick followup: `@sanity/table` it is - meaning, we will take it in officially for v3.

Are you ok with me picking up this PR later this week, moving a fork into the sanity org and publishing it?
If you would rather us hash it out here, I'm all for it.

Thank you for doing this :D 🥇

**Comment by davydog187** [comment]
Hello @snorrees, I appreciate the writeup! That all sounds great to me, and I can certainly test out the `plugin-kit` features you mentioned, although I would love for sanity to take over stewardship for this plugin to V3!

Another item that is important to me, we want to add features to this table over time, including PortableText support for table cells. Is that something you see supported in the official Sanity fork? Would you accept PRs for that?

Regarding the Sanity Meetup, I will try to attend tomorrow, I signed up! Will you be talking about this plugin, or more generally about `plugin-kit`, or both? Looking forward to it

**Comment by snorrees** [comment]
Will be about plugin-kit, its not a very long piece, but a intro to how we encourage it as the one-stop-shop for building sanity packages. Zero-to-npm-published in the time of ☕ 🤞 

If you are looking to add a bunch of features the quickest way forward for you will probably be to maintain it yourself. 

We will of course try to be on the ball on looking at PR's but I cannot make any promises unfortunately. 
This time around we will put this out as a version for V3, based on this PR and revisit as time permits.
At least this way, there is a much shorter path from @sanity/table to your feature-added fork, hopefully.

**Comment by snorrees** [comment]
@davydog187 [@sanity/table 1.0.0 ](https://github.com/sanity-io/table/releases/tag/v1.0.0)for v3 is now out. Thanks a lot for this work!

**Pull request merged**